internal/types/document.go

```diff
@@ -428,6 +430,38 @@ func (d *Document) moveIDToTheFirstIndex() {
 	d.fields = slices.Delete(d.fields, idIdx+1, idIdx+2)
 }
 
+// Iterator returns an iterator over the document fields.
+func (d *Document) Iterator() iterator.Interface[string, any] {
+	return newDocumentIterator(d)
+}
+
+// documentIterator represents an iterator over the document fields.
+type documentIterator struct {
+	doc         *Document
+	currentIter atomic.Uint32
```

Comment:
Do we need `currentInter` at all? If we work with field names, I don't think we need this number.